workers/python/kitchen_sink.py

```diff
@@ -140,25 +184,44 @@ async def handle_action(
             workflow.set_query_handler(action.query_handler.name, workflow_query)
             return (True, None)
         elif action.signal:
-            queue = self._signal_recieved.setdefault(
-                action.signal.name, asyncio.Queue()
-            )
-            await queue.get()
-            queue.task_done()
+            signal_event = asyncio.Event()
+
+            def signal_handler(arg: Optional[Any] = None) -> None:
+                signal_event.set()
+
+            workflow.set_signal_handler(action.signal.name, signal_handler)
+            await signal_event.wait()
```

Comment:
Is there a reason you don't return in this branch?

Reply:
Same reason I don't in sleep and we shouldn't in many other places we do. No need to repeat `return (False, None)` in each arm.